apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -328,9 +327,18 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
       invokeActionMessages(response.content);
 
       trpcUtils.chat.appSessions.getChat.invalidate({ id });
+      clearHashCache();
```

Comment:
Is the onFinish handler reliable? Do we need to have similar logic in onError?

Reply:
Interesting, I hadn't considered that